build-aux/snap/snapcraft.yaml

```diff
@@ -28,172 +29,294 @@ license: GPL-3.0
+
+      cd "${CRAFT_PART_BUILD}"
+
+      # Work-around for snapcraft not working with git worktrees
+      # We detect it with .git being a file rather than a directory
+      if [ -f .git ]; then
+        EXTRA_GO_FLAGS="-buildvcs=false"
+      fi
+      CMDS=(bin/snap
+            lib/snapd/snapd
+            lib/snapd/snap-exec
+            lib/snapd/snap-failure
+            lib/snapd/snap-fde-keymgr
+            lib/snapd/snap-preseed
+            lib/snapd/snap-recovery-chooser
+            lib/snapd/snap-repair
+            lib/snapd/snap-seccomp
+            lib/snapd/snap-update-ns
+            lib/snapd/snapctl
+            lib/snapd/snapd-apparmor)
+      for cmd in "${CMDS[@]}"; do
+        case "${cmd}" in
+           # FIXME: some binaries need to be run confined in apps. But
+           # instead we should allow apps to access dynamic linkers
+           # and libraries from snapd.
+           lib/snapd/snap-exec|lib/snapd/snapctl)
+             export CGO_ENABLED=0
+             GO_LD_FLAGS=()
+             ;;
+           lib/snapd/snap-update-ns)
+             export CGO_ENABLED=1
+             GO_LD_FLAGS=(--ldflags '-extldflags "-static"')
+             ;;
+           *)
+             export CGO_ENABLED=1
+             GO_LD_FLAGS=()
+             ;;
+        esac
+
+        case "${cmd}" in
+          bin/snap)
+            TAGS=(nomanagers)
+            case "$(craftctl get version)" in
+              1337.*)
+                TAGS+=(withtestkeys faultinject)
+                ;;
+            esac
+            ;;
+          *)
+            TAGS=()
+            case "$(craftctl get version)" in
+              1337.*)
+                TAGS+=(withtestkeys withbootassetstesting faultinject)
+                ;;
+            esac
+            ;;
+        esac
+
+        go build -mod=vendor -tags "${TAGS[*]}" "${GO_LD_FLAGS[@]}" ${EXTRA_GO_FLAGS-} -o "${CRAFT_PART_INSTALL}/usr/${cmd}" "github.com/snapcore/snapd/cmd/$(basename ${cmd})"
+        unset CGO_ENABLED
+      done
+
+      make -C data -j"${CRAFT_PARALLEL_BUILD_COUNT}" all
+      # wrappers/core18.go expects units in fixed paths so even when
+      # systemd package will those, they have to continue to match our
+      # paths. We force the value here in case we change the default.
+      make -C data install DESTDIR="${CRAFT_PART_INSTALL}" SYSTEMDUSERUNITDIR=/usr/lib/systemd/user SYSTEMDSYSTEMUNITDIR=/lib/systemd/system
+      # UC depends on this name (symlink)
+      mv "${CRAFT_PART_INSTALL}/etc/profile.d/snapd.sh" "${CRAFT_PART_INSTALL}/etc/profile.d/apps-bin-path.sh"
+
+      install -Dm644 data/info "${CRAFT_PART_INSTALL}/usr/lib/snapd/info"
+      install -Dm644 data/preseed.json "${CRAFT_PART_INSTALL}/usr/lib/snapd/preseed.json"
+
+      install -Dm644 -t "${CRAFT_PART_INSTALL}/usr/lib/snapd" data/completion/bash/complete.sh
+      install -Dm644 -t "${CRAFT_PART_INSTALL}/usr/share/bash-completion/completions" ./data/completion/bash/snap
```

Comment:
info and completions aren't supported on UC, so we need preseed.json, but likely not the other files.

Reply:
Completions work on UC.

Reply:
The info file is used for `snap/info.go`. I do think we need it.

Reply:
Uh, right, ignore that, I'm a muppet. That's the snapd info file we use internally.